inst/include/Rcpp/exceptions.h

```diff
@@ -127,17 +127,17 @@ namespace Rcpp{
 } // namespace Rcpp
 
 inline SEXP get_last_call(){
-    SEXP sys_calls_symbol = Rf_install( "sys.calls" ) ;
-    Rcpp::Shield<SEXP> sys_calls_expr( Rf_lang1(sys_calls_symbol) );
+    SEXP sys_calls_symbol = Rf_install( "sys.call" ) ;
+
+    // -9 Skips the wrapped tryCatch from Rcpp_eval
+    Rcpp::Shield<SEXP> sys_calls_expr( Rf_lang2(sys_calls_symbol, Rf_ScalarInteger(-9)) );
```

Comment:
At the last R Summit Luke warned against hard coding the number of calls on the stack so I think this might indeed fail in the presence of said eager byte-compilation.